test/detection/test_line_counter.py

```diff
@@ -70,3 +73,320 @@ def test_calculate_region_of_interest_limits(
+            [
+                [[100, 50, 120, 70], [100, 50, 120, 70]],
+                [[-100, 50, -80, 70], [100, 50, 120, 70]],
+                [[100, 50, 120, 70], [100, 50, 120, 70]],
+            ],
+            [[False, False], [False, False], [True, False]],
+            [[False, False], [True, False], [False, False]],
+        ),
+        (
+            Vector(
+                Point(0, 0),
+                Point(-100, 0),
+            ),
+            [
+                [[-50, 70, -40, 50], [-80, -50, -70, -40]],
+                [[-50, -70, -40, -50], [-80, 50, -70, 40]],
+                [[-50, 70, -40, 50], [-80, 50, -70, 40]],
+                [[-50, -70, -40, -50], [-80, 50, -70, 40]],
+                [[-50, 70, -40, 50], [-80, 50, -70, 40]],
+                [[-50, -70, -40, -50], [-80, 50, -70, 40]],
+                [[-50, 70, -40, 50], [-80, 50, -70, 40]],
+                [[-50, -70, -40, -50], [-80, -50, -70, -40]],
+            ],
+            [
+                (False, False),
+                (False, True),
+                (True, False),
+                (False, False),
+                (True, False),
+                (False, False),
+                (True, False),
+                (False, False),
+            ],
+            [
+                (False, False),
+                (True, False),
+                (False, False),
+                (True, False),
+                (False, False),
+                (True, False),
+                (False, False),
+                (True, True),
+            ],
+        ),
+    ],
+)
+def test_line_zone_multiple_detections(
+    vector, bbox_sequence, expected_count_in: List[bool], expected_count_out: List[bool]
+) -> None:
+    line_zone = LineZone(start=vector.start, end=vector.end)
+    for i, bboxes in enumerate(bbox_sequence):
+        detections = mock_detections(
+            xyxy=bboxes,
+            tracker_id=[i for i in range(0, len(bboxes))],
+        )
+        count_in, count_out = line_zone.trigger(detections)
+        assert np.all(count_in == expected_count_in[i])
+        assert np.all(count_out == expected_count_out[i])
+
+
+@pytest.mark.parametrize(
+    "vector, bbox_sequence",
+    [
+        (
+            Vector(
+                Point(0, 0),
+                Point(0, 100),
+            ),
+            [
+                [100, 50, 120, 70],
+                [-100, 50, -80, 70],
+            ],
+        ),
+        (
+            Vector(
+                Point(0, 0),
+                Point(0, 100),
+            ),
+            [
+                [-100, 50, -80, 70],
+                [100, 50, 120, 70],
+            ],
+        ),
+    ],
+)
+def test_line_zone_does_not_count_detections_without_tracker_id(vector, bbox_sequence):
+    line_zone = LineZone(start=vector.start, end=vector.end)
+    for bbox in bbox_sequence:
+        detections = Detections(
+            xyxy=np.array([bbox]).reshape((-1, 4)),
+            tracker_id=np.array([None]),
```

Comment:
This was a loop of `[None for x in range(0,1)]`, so we correctly simplified to `[None]`.

However, is this right? I think the length of `tracker_id` does not match `xyxy`.
Should we change this to `tracker_id=[i for i in range(0, len(bbox))],`?

Reply:
I'd recommend merging `test_line_zone_does_not_count_detections_without_tracker_id` with `test_line_zone_single_detection`. Case is almost the same.

Reply:
@SkalskiP it's OK - test's input is a sequence of positions of single tracklet, each position described by 4 ints (which are xyxy of the bounding box) .

Reply:
@SkalskiP I have mixed feelings about this. I understand your point that the inputs and code are the same, but this test tests different functionality. LinasKo claims that occurrences of 'None' 'tracker_id' are an unconfirmed anomaly that should be eradicated. Maybe it's better to leave it as a separate test in case some day it becomes settled that 'tracker_id' can't be None (and hence this test becomes obsolete and fails). But I leave this decision up to you.

Reply:
Hi @tc360950,

It was bothering me so I verified it - values inside `tracker_id` are always integers and never `None`. Please remove the tests testing for `None` 😉

Reply:
Agree. As far as I know, per box, `tracker_id` should not be `None`. 

But I meant to merge those two suits and potentially use the `exception` argument to handle cases that throw. You can find an example of this in [`test_calculate_region_of_interest_limits`](https://github.com/roboflow/supervision/blob/1bbe03ca27dda82b8910feb0352dfb3a443f970d/test/detection/test_line_counter.py#L65). We adopted this convention in other unit tests in `supervision` and I would like to maintain it.

Reply:
@SkalskiP since there's an open PR by LinasKo which removes check for 'None' 'tracker_id' I've deleted this test.